server/models/UserModel/index.js

```diff
@@ -35,21 +35,22 @@ const UserModel = {
 
     return await db.collection(collections.users).findOne({ _id: insertedId })
   },
-  findOne: async (db, userAttributes) => {
-    const user = await db
-      .collection(collections.users)
-      .findOne(userAttributes, {
-        projection: userMongoProjection,
-      })
+  findOne: async (db, userAttributes, includeFields = {}) => {
+    const includeOrExcludeFields = Object.keys(includeFields)?.length
+      ? includeFields
+      : userMongoProjection
 
-    return user
+    return await db.collection(collections.users).findOne(userAttributes, {
+      projection: includeOrExcludeFields,
+    })
   },
   update: async (db, dataDb, uid, userUpdates) => {
```

Comment:
Can we add a check here that the user is either an admin or updating their own account? Also if they are a non-admin they shouldn't be able to change their own role.

Reply:
I've added check if a user is an admin and if they are modifying their own account 
[53b0a68](https://github.com/AMP-SCZ/dpdash/pull/563/commits/53b0a68b784fb09ff54e66de93994b268c4daa8c)

Reply:
Added a comment, sorry I was unclear. This change isn't what I was trying to get at. Hit me up if you want to sync.